amads/melody/contour/interpolation_contour.py

```diff
@@ -0,0 +1,272 @@
+
+        Returns
+        -------
+        list[int]
+            List containing the signs (+1, 0, -1) of the gradient values in the contour
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.direction[:4]  # Show first 4 values
+        [-1, -1, -1, -1]
+        """
+        # Calculate the signum of the interpolation contour using numpy's sign function
+        global_direction = np.sign(self.contour)
+
+        return [int(x) for x in global_direction]
+
+    @property
+    def mean_gradient(self) -> float:
+        """Calculate the absolute mean gradient of the interpolation contour.
+
+        Returns
+        -------
+        float
+            Mean of the absolute gradient values
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.mean_gradient
+        30.0
+        """
+        return float(np.mean(np.abs(self.contour)))
+
+    @property
+    def gradient_std(self) -> float:
+        """Calculate the standard deviation of the interpolation contour gradients.
+
+        Returns
+        -------
+        float
+            Standard deviation of the gradient values
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.gradient_std
+        25.9272...
+        """
+        return float(np.std(self.contour))
+
+    @property
+    def direction_change_ratio(self) -> float:
+        """Calculate the ratio of direction changes in the interpolation contour.
+
+        This measures the number of changes in contour direction relative to the number
+        of interpolation lines (i.e. number of different gradient values).
+
+        Returns
+        -------
+        float
+            Ratio of direction changes to total gradient changes (between 0 and 1)
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3, 0.4], [60, 58, 64, 65, 63])
+        >>> ic.direction_change_ratio
+        2.0
+        """
+        # Get signs of all gradients
+        signs = np.sign(self.contour)
+
+        # Count direction changes (where consecutive signs are different)
+        direction_changes = np.sum(signs[:-1] != signs[1:])
+
+        # Count total gradient changes (where consecutive values are different)
+        total_changes = np.sum(self.contour[:-1] != self.contour[1:])
+
+        # Avoid division by zero
+        if total_changes == 0:
+            return 0.0
+
+        return float(direction_changes / total_changes)
+
+    @property
+    def class_label(self) -> str:
+        """Classify an interpolation contour into gradient categories.
+
+        The contour is sampled at 4 equally spaced points and each gradient is
+        normalized and classified into one of 5 categories:
+
+        - 'a': Strong downward (-2) - normalized gradient <= -1.45
```

Comment:
You need to give the units of normalized gradient

Reply:
added